target/linux/rockchip/rk3399/target.mk

```diff
@@ -0,0 +1,9 @@
+ARCH:=aarch64
+SUBTARGET:=rk3399
+BOARDNAME:=RK3399 boards (64 bit)
+CPU_TYPE:=cortex-a53
```

Comment:
Drop the `CPU_TYPE`, this will allow for multiple ARM64 core types to be supported from single target now and probably in the future. This mistake has already been made for mvebu which puts additional strain on buildbots.

Reply:
Why would we drop CPU_TYPE?

Reply:
Did You read second sentence?

Reply:
I don't follow here as well:

```
target/linux/armvirt/64/target.mk  ARCH:=aarch64
target/linux/bcm27xx/bcm2710/target.mk CPU_TYPE:=cortex-a53 ARCH:=aarch64
target/linux/bcm27xx/bcm2711/target.mk CPU_TYPE:=cortex-a72 ARCH:=aarch64
target/linux/ipq807x/Makefile CPU_TYPE:=cortex-a53 ARCH:=aarch64
target/linux/layerscape/armv8_64b/target.mk  ARCH:=aarch64
target/linux/mediatek/mt7622/target.mk CPU_TYPE:=cortex-a53 ARCH:=aarch64
target/linux/mvebu/cortexa53/target.mk CPU_TYPE:=cortex-a53 ARCH:=aarch64
target/linux/mvebu/cortexa72/target.mk CPU_TYPE:=cortex-a72 ARCH:=aarch64
target/linux/octeontx/Makefile  ARCH:=aarch64
target/linux/rockchip/rk3399/target.mk CPU_TYPE:=cortex-a53 ARCH:=aarch64
target/linux/sunxi/cortexa53/target.mk CPU_TYPE:=cortex-a53 ARCH:=aarch64
```

Reply:
What does dropping CPU_TYPE do? Does it use the lowest instruction set supported by all aarch64 CPUs? Or something else?

Reply:
@ynezz Look at my answer to blocktrron, it's always best to support wide spectrum of SoCs with single target, which would have benefit of less strain on buldbots and download server.

@Manawyrm Yes.

Reply:
> which would have benefit of less strain on buldbots and download server

@tmn505 But as you can see from my list above, it makes no sense to me. We've already plenty of aarch64 targets which use CPU_TYPE to enable GCC performance tuning already. So what would we gain by removing CPU_TYPE in this target? Probably just attract hordes of benchmark freaks, which would run in the circles and screaming something about intentional crippled performance of their CPUs :-)

Reply:
@ynezz Yes, we do have them. For example rk1808 is Cortex A35 would need additional subtarget, where toolchain and packages would need to be built from ground up. With dropping CPU_TYPE you could potentially support it under the same target. You can't please everyone, they do complain but at the end of the day part of them still use Debian or other main distro, which doesn't have that fine tuning. If the hordes want the fine tuning, there's nothing against adding optimizations to local build. BTW. I'm more afraid of jow- than the hordes :-)

Reply:
> rk1808 is Cortex A35
> With dropping CPU_TYPE you could potentially support it under the same target.

Ok, now this makes sense.

Reply:
I have dropped the CPU_TYPE.

Reply:
@tmn505 @ynezz 
If that's the case we can add it to new (unique) subtargets, there's no point in doing to for already existing ones and if we're going to use Debian as an example, they don't as far as I kno enforce -Os either if you care about performance which many seems to do...